lib/backend/etcd/syncer.go

```diff
@@ -278,6 +279,13 @@ func (syn *etcdSyncer) watchEtcd(watcherUpdateC chan<- interface{}) {
 			// Wait for the next event from the watcher.
 			resp, err := watcher.Next(context.Background())
 			if err != nil {
+				// Prevent a tight loop if etcd is repeatedly failing.
+				if time.Since(timeOfLastError) < 1*time.Second {
+					log.Warning("May be tight looping, throttling retries.")
+					time.Sleep(1 * time.Second)
+				}
+				timeOfLastError = time.Now()
+
 				if !retryableWatcherError(err) {
```

Comment:
Yes, that's what I wanted to guard against.  Since we're comparing with the last error time we'll only trigger the sleep if we hit two errors within 1s.